chronos/scripts/upload_chronos_gr_config

```diff
@@ -0,0 +1,71 @@
+
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2016  Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+FILENAME=/etc/chronos/chronos_gr.conf
+local_site_name=site1
+etcd_key=clearwater
+. /etc/clearwater/config
+
+# Check we can contact `etcd`
+if ! nc -z ${management_local_ip:-$local_ip} 4000
+then
+  echo "The Clearwater Configuration store (etcd) is not running"
+  echo "Please start it before uploading configuration"
+  exit 2
+fi
+
+# Check that there's chronos GR configuration to upload
+if [[ ! -f $FILENAME ]]
+then
+  echo "No Chronos GR configuration detected"
+  exit 2
+fi
+
+# Upload the file to etcd
+keypath=http://${management_local_ip:-$local_ip}:4000/v2/keys/$etcd_key/$local_site_name/configuration/chronos_gr_config
+curl -X PUT $keypath --data-urlencode value@$FILENAME 2> /tmp/upload-chronos-gr-config.stderr.$$ | tee /tmp/upload-chronos-gr-config.stdout.$$ | egrep -q "\"action\":\"set\""
+rc=$?
+
+# Check the return code and log if appropriate.
+if [ $rc != 0 ] ; then
+  echo Upload chronos GR configuration failed to $keypath  >&2
+  cat /tmp/upload-chronos-gr-config.stderr.$$              >&2
+  cat /tmp/upload-chronos-gr-config.stdout.$$              >&2
+fi
+rm -f /tmp/upload-chronos-gr-config.stderr.$$ /tmp/upload-chronos-gr-config.stdout.$$
```

Comment:
It feels like the lines above should be something that would be in a common function... in a sensible language (i.e. not bash).